src/pages/examples/DrawingToolsPage.tsx

```diff
@@ -0,0 +1,283 @@
+import ViewportIcons from "../../components/shapediver/viewport/ViewportIcons";
+import { useSessionPropsParameter } from "hooks/shapediver/parameters/useSessionPropsParameter";
+import AcceptRejectButtons from "../../components/shapediver/ui/AcceptRejectButtons";
+import useAppBuilderSettings from "hooks/shapediver/useAppBuilderSettings";
+import TabsComponent, { ITabsComponentProps } from "components/ui/TabsComponent";
+import { IconTypeEnum } from "types/shapediver/icons";
+import { IAppBuilderSettingsSession } from "types/shapediver/appbuilder";
+import useDefaultSessionDto from "hooks/shapediver/useDefaultSessionDto";
+import { useDefineGenericParameters } from "hooks/shapediver/parameters/useDefineGenericParameters";
+import { ShapeDiverResponseParameterType } from "@shapediver/api.geometry-api-dto-v2";
+import { IGenericParameterDefinition } from "types/store/shapediverStoreParameters";
+import { CustomizationProperties, PlaneRestrictionApi } from "@shapediver/viewer.features.drawing-tools";
+import { useDrawingTools } from "hooks/shapediver/viewer/useDrawingTools";
+import { useDrawingToolsApiStore } from "store/useDrawingToolsApiStore";
+import { IGeometryData } from "@shapediver/viewer";
+import { useParameterStateless } from "hooks/shapediver/parameters/useParameterStateless";
+import { useOutputContent } from "hooks/shapediver/viewer/useOutputContent";
+
+const VIEWPORT_ID = "viewport_1";
+
+interface Props extends IAppBuilderSettingsSession {
+	/** Name of example model */
+	example?: string;
+}
+
+/**
+ * Function that creates the drawing tools page.
+ * The aside (right side) tabs with a ParameterUiComponent
+ * and a viewport and session in the main component. The session is connected via its id to the ParameterUiComponent.
+ *
+ * @returns
+ */
+export default function DrawingToolsPage(props: Partial<Props>) {
+
+	const { defaultSessionDto } = useDefaultSessionDto(props);
+	const { settings } = useAppBuilderSettings(defaultSessionDto);
+	const sessionCreateDto = settings ? settings.sessions[0] : undefined;
+	const sessionId = sessionCreateDto?.id ?? "";
+
+	// use a session with a ShapeDiver model and register its parameters
+	const { sessionApi } = useSession(sessionCreateDto);
+	useEffect(() => {
+		if (sessionApi)
+			console.debug(`Available output names: ${Object.values(sessionApi.outputs).map(o => o.name)}`);
+	}, [sessionApi]);
+
+	// get parameters that don't have a group or whose group name includes "export"
+	const parameterProps = useSessionPropsParameter(sessionId, param => !param.group || !param.group.name.toLowerCase().includes("export"));
+
+	/////
+	// START - Example on how to apply the drawing tools
+	/////
+
+	// state for the drawing tools application
+	const [outputNameDrawingTools, setOutputNameDrawingTools] = useState<string>("");
+
+	// get the parameter API
+	const parameter = useParameterStateless<string>(sessionId, "points");
```

Comment:
@MajorMeerkatThe3rd as an optimization you could save the parameter using a useRef which you use from the onUpdate callback, which means you won't need a dependency for the onUpdate callback